kafka-streams-partitioners/weighted-group-partitioner/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/KeyHashPartitioner.java

```diff
@@ -0,0 +1,18 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+
+public class KeyHashPartitioner<K, V> implements StreamPartitioner<K, V> {
+  private final AtomicInteger currentPartition = new AtomicInteger();
+
+  @Override
+  public Integer partition(String topic, K key, V value, int numPartitions) {
+    int hashcode = 0;
```

Comment:
Not entirely sure where we plan to use this but I'm wondering if this hashCode based partitioning will distribute well.
Shouldn't we try to mimic the DefaultPartitioner. By that I mean, we create a Partitioner that accepts a KeySerializer which is used to convert the K to a byte[] which can be passed to the murmur2 Utils function - `Utils.toPositive(Utils.murmur2(keyBytes)) % numPartitions`